### PR TITLE
feat: custom and config enable view unread message

### PR DESCRIPTION
**Pull request by huydosgtech**

### CODE REVIEW

File: src/chat/components/ViewUnRead.tsx [hunk 1]
```diff
@@ -22,8 +23,11 @@ const ViewUnRead: React.FC<ViewUnReadProps> = ({
   customTextStyle,
   unReadSentMessage = 'Sent',
   unReadSeenMessage = 'Seen',
+  viewUnReadCustom,
 }) => {
-  return (
+  return viewUnReadCustom ? (
+    viewUnReadCustom()
```
Comment (loc-nguyenthien) [inline src/chat/components/ViewUnRead.tsx:29]:
```suggestion
    viewUnReadCustom(userUnreadMessage)
```
Link to the above comment

File: src/chat/components/ViewUnRead.tsx [hunk 2]
```diff
@@ -14,6 +14,7 @@ interface ViewUnReadProps {
   customTextStyle?: StyleProp<TextStyle>;
   unReadSentMessage?: string;
   unReadSeenMessage?: string;
+  viewUnReadCustom?: () => JSX.Element;
```
Comment (loc-nguyenthien) [inline src/chat/components/ViewUnRead.tsx:17]:
```suggestion
  viewUnReadCustom?: (hasUnread: boolean) => JSX.Element;
```
I suggest we pass in the `userUnreadMessage` prop into this custom component to allow easier customize the UI with unread/read status

File: src/chat/components/bubble/CustomBubble.tsx [hunk 3]
```diff
@@ -18,6 +18,8 @@ interface CustomBubbleProps {
   customTextStyle?: StyleProp<ViewStyle>;
   unReadSentMessage?: string;
   unReadSeenMessage?: string;
+  enableSeenMessage: boolean;
+  viewUnReadCustom?: () => JSX.Element;
```
Comment (loc-nguyenthien) [inline src/chat/components/bubble/CustomBubble.tsx:22]: Similar above comment

File: src/chat/components/bubble/CustomBubble.tsx [hunk 4]
```diff
@@ -30,25 +32,42 @@ export const CustomBubble: React.FC<CustomBubbleProps> = ({
...
 
-  const renderBubble = (currentMessage: MessageProps) => {
-    const isMyLatestMessage =
-      !Object.keys(bubbleMessage.nextMessage ?? {}).length &&
-      position === 'right';
-    const ViewRead = isMyLatestMessage && (
+  const viewUnRead = (
+    isMyLatestMsg: boolean,
+    enableSeenMsg: boolean,
+    viewUnReadCustomUI?: () => JSX.Element
```
Comment (loc-nguyenthien) [inline src/chat/components/bubble/CustomBubble.tsx:46]: Similar above comment

**Review by loc-nguyenthien — CHANGES_REQUESTED** [review]

File: src/chat/components/ViewUnRead.tsx (hunk 1 above)
Comment (huydosgtech) [reply src/chat/components/ViewUnRead.tsx]: ok I updated

File: src/chat/components/ViewUnRead.tsx (hunk 2 above)
Comment (huydosgtech) [reply src/chat/components/ViewUnRead.tsx]: ok I updated

**Review by loc-nguyenthien — APPROVED** [review]

File: src/chat/ChatScreen.tsx [hunk 5]
```diff
@@ -63,6 +63,8 @@ interface ChatScreenProps extends GiftedChatProps {
   timeoutSendNotify?: number;
   enableTyping?: boolean;
   typingTimeoutSeconds?: number;
+  enableSeenMessage?: boolean;
+  viewUnReadCustom?: (hasUnread: boolean) => JSX.Element;
```
Comment (danhmaisgt) [inline src/chat/ChatScreen.tsx:67]: I think the name of this prop should be `customUnReadView` or `customViewUnRead`

File: src/chat/components/ViewUnRead.tsx [hunk 6]
```diff
@@ -14,6 +14,7 @@ interface ViewUnReadProps {
   customTextStyle?: StyleProp<TextStyle>;
   unReadSentMessage?: string;
   unReadSeenMessage?: string;
+  viewUnReadCustom?: (hasUnread: boolean) => JSX.Element;
```
Comment (danhmaisgt) [inline src/chat/components/ViewUnRead.tsx:17]: I think the name of this prop should be `customUnReadView` or `customViewUnRead`

File: src/chat/components/bubble/CustomBubble.tsx [hunk 7]
```diff
@@ -18,6 +18,8 @@ interface CustomBubbleProps {
   customTextStyle?: StyleProp<ViewStyle>;
   unReadSentMessage?: string;
   unReadSeenMessage?: string;
+  enableSeenMessage: boolean;
+  viewUnReadCustom?: (hasUnread: boolean) => JSX.Element;
```
Comment (danhmaisgt) [inline src/chat/components/bubble/CustomBubble.tsx:22]: I think the name of this prop should be `customUnReadView` or `customViewUnRead`

Comment (huydosgtech) [reply src/chat/components/bubble/CustomBubble.tsx]: ok I renamed it

**Review by danhmaisgt — APPROVED** [review]

File: src/chat/components/bubble/CustomBubble.tsx [hunk 8]
```diff
@@ -30,25 +32,42 @@ export const CustomBubble: React.FC<CustomBubbleProps> = ({
...
     left: { backgroundColor: 'transparent' },
     right: { backgroundColor: 'transparent' },
   };
 
-  const renderBubble = (currentMessage: MessageProps) => {
-    const isMyLatestMessage =
-      !Object.keys(bubbleMessage.nextMessage ?? {}).length &&
-      position === 'right';
-    const ViewRead = isMyLatestMessage && (
+  const viewUnRead = (
```
Comment (loc-nguyenthien) [inline src/chat/components/bubble/CustomBubble.tsx:43]:
```suggestion
  const renderViewUnRead = (
```
Rename this for clarification

**Review by loc-nguyenthien — CHANGES_REQUESTED** [review]

**Review by loc-nguyenthien — APPROVED** [review]

**Review by hoangSTS — APPROVED** [review]

File: src/chat/ChatScreen.tsx [hunk 9]
```diff
@@ -63,6 +63,8 @@ interface ChatScreenProps extends GiftedChatProps {
   timeoutSendNotify?: number;
   enableTyping?: boolean;
   typingTimeoutSeconds?: number;
+  disableSeenMessage?: boolean;
```
Comment (honghoangsts) [inline src/chat/ChatScreen.tsx:66]: messageStatusEnable

File: src/chat/ChatScreen.tsx [hunk 10]
```diff
@@ -328,6 +331,8 @@ export const ChatScreen: React.FC<ChatScreenProps> = ({
         customTextStyle={props.customTextStyle}
         unReadSentMessage={props.unReadSentMessage}
         unReadSeenMessage={props.unReadSeenMessage}
+        customUnReadView={props.customUnReadView}
```
Comment (honghoangsts) [inline src/chat/ChatScreen.tsx:334]: customMessageStatus

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: src/chat/ChatScreen.tsx (hunk 9 above)
Comment (huydosgtech) [reply src/chat/ChatScreen.tsx]: I updated

File: src/chat/ChatScreen.tsx (hunk 10 above)
Comment (huydosgtech) [reply src/chat/ChatScreen.tsx]: I updated

**Review by honghoangsts — APPROVED** [review]

**Pull request merged**